### ISSUE TITLE
Separator shouldn't display at bottom of "File" menu on Mac OS X

### ISSUE THREAD

**Issue by follower** [LMMS/lmms#3345]
This seems to be a regression related to the move from Qt4 to Qt5.

Notice there is a separator at the bottom of the "File" menu which is unnecessary and shouldn't be there:
![lmms-menu-separator-issue-qt5](https://cloud.githubusercontent.com/assets/189962/22863154/250bb8d8-f1a0-11e6-9d93-f3b2273fe01d.png)

Version information (this is on Mac OS X 10.8.x):
![lmms-menu-separator-issue-qt5-version-info](https://cloud.githubusercontent.com/assets/189962/22863157/2519e994-f1a0-11e6-967d-474ac9aba158.png)

For comparison here is the same menu with Qt4:
![lmms-menu-separator-issue-qt4](https://cloud.githubusercontent.com/assets/189962/22863155/250c2ca0-f1a0-11e6-8ff4-b4c98b3c99ed.png)

And version information for Qt4 build on same system:
![lmms-menu-separator-issue-qt4-version-info](https://cloud.githubusercontent.com/assets/189962/22863156/251787b2-f1a0-11e6-862c-5bbe07f932ee.png)

The reason this occurs is because Qt automatically moves the "Quit" menu item from the file menu to the "~~Apple~~ Application Menu" as is usual for the platform. Normally any redundant separator is also removed at the same time.

This appears to be a known issue for Qt: ["[QTBUG-40071] separatorsCollapsible does not work when menu items are moved to the application menu on Mac OS X"](https://bugreports.qt.io/browse/QTBUG-40071).

I've added further detail to the QTBUG entry but have been unable to determine an underlying cause so far.

Before I realised it was an issue with Qt I developed a simple workaround which results in:

![lmms-menu-separator-issue-qt5-workaround](https://cloud.githubusercontent.com/assets/189962/22863264/a16d6240-f1a1-11e6-9c9b-39099602f67a.png)

Given that it appears to be a known issue that's not been resolved in nearly 3 years I'd suggest using the workaround in lmms in the meantime.

If desired I can make a PR with the workaround (with further documentation):

```
diff --git a/src/gui/MainWindow.cpp b/src/gui/MainWindow.cpp
index 4a5bacb..aa943c8 100644
--- a/src/gui/MainWindow.cpp
+++ b/src/gui/MainWindow.cpp
@@ -315,7 +315,9 @@ void MainWindow::finalize()
                                        SLOT( exportProjectMidi() ),
                                        Qt::CTRL + Qt::Key_M );*/
 
+#ifndef LMMS_BUILD_APPLE
        project_menu->addSeparator();
+#endif
        project_menu->addAction( embed::getIconPixmap( "exit" ), tr( "&Quit" ),
                                        qApp, SLOT( closeAllWindows() ),
                                        Qt::CTRL + Qt::Key_Q );

```

**Comment by tresf** [comment]
@follower please issue a pull request and we'd happily merge.  Also; if Qt5 specific also add `#if QT_VERSION >= 0x050000 ...`

**Comment by follower** [comment]
Thanks for taking a look.

I'm working on a PR and decided it was probably still wise to keep the version check as we know the workaround is not needed on 4 and my impression of Qt so far is that assumptions it won't break anything on Qt 4 may be misplaced. :)

Also, I've just realised that the "Help" menu also has the same issue, so will do the same there.

**Comment by follower** [comment]
The original Qt bug has now been closed as "Cannot Reproduce" as of Qt 5.6, are you able to confirm this with one of your builds with a later Qt @tresf?

(Side note: As you might've guessed I finally managed to get Qt 5.5.1 + lmms building okay, so hopefully I can now contribute something more to lmms than requests for builds for obsolete OS. :D )

**Comment by tresf** [comment]
>  are you able to confirm [`QTBUG-40071` is fixed] with one of your builds with a later Qt @tresf?

Yes, confirmed as fixes in Qt 5.7

![image](https://cloud.githubusercontent.com/assets/6345473/22986954/36278548-f37b-11e6-9ada-48d033695625.png)

**Comment by follower** [comment]
So that's fixed within Qt 5.7 and without needing a workaround in lmms?

If that's the case, would you like me to add an upper bound to the Qt version check for the workaround?

**Comment by tresf** [comment]
> So that's fixed within Qt 5.7 and without needing a workaround in lmms?

Sounds like a noble idea.  We're probably releasing 1.2 against Qt5.5 on Mac due to the [`macdeployqt` bug](https://github.com/LMMS/lmms/pull/2861) that prevents us from packaging on newer versions (I'm currently working around this [here](https://github.com/tresf/lmms/commit/f24f3d94bf3e9400003ebb962a77718f020a642f) but haven't merged the changes in yet.

**Comment by follower** [comment]
I guess I'll take Qt's word for it and make the check be >= 5.0 and < 5.6. Feel free to verify 5.6. :)

**Comment by follower** [comment]
Closed by merge of #3350.

**Comment by tresf** [comment]
> This suggests the upper-bound version check should probably just be removed. 

Agreed.  I see no harm in this. 👍
